RGBMatrixEmulator/graphics/__init__.py

```diff
@@ -11,26 +11,37 @@ def DrawText(canvas, font, x, y, color, text):
     if len(text) == 0:
         return
 
-    # Ensure text doesn't get drawn as multiple lines
-    linelimit = len(text) * (font.bdf_font.headers['fbbx'] + 1)
-
-    # TODO: This is VERY slow for large text
-    # See mlb-led-scoreboard offday renderer with headlines
-    text_map = font.bdf_font.draw(text, linelimit).todata(2)
-    font_y_offset = -(font.bdf_font.headers['fbby'] + font.bdf_font.headers['fbbyoff'])
-
-    for y2, row in enumerate(text_map):
-        for x2, value in enumerate(row):
-            if value == 1:
-                try:
-                    if isinstance(color, tuple):
-                        canvas.SetPixel(x + x2, y + y2 + font_y_offset, *color)
-                    else:
-                        canvas.SetPixel(x + x2, y + y2 + font_y_offset, color.r, color.g, color.b)
-                except Exception:
-                    pass
-
-    return len(text_map[0])
+    text_orig = text
+
+    # crop text to increase speed dramatically
+    charwidth = int(font.bdf_font.props["cap_height"])
```

Comment:
I believe this is the height, not width. You'll probably want to use the font bounding box: `font.bdf_font.headers['fbby']`

Might also need the offset? `font.bdf_font.headers['fbbxoff']`

See the spec for `bdfparser`: https://font.tomchen.org/bdf_spec/file_format/global_font_info#fontboundingbox

Reply:
I'll make this change. 

In general, it seems the various bdf fonts I've encountered are inconsistent with whether or not their 'bounding box' declared size includes any space pixels - for example, the default font for mlb-led-scoreboard is '4x6', but lists its bounding box as 3 6 0 -1. Another font used, a 5x8, has a bounding box of 5 8 0 -1. This change may therefore be a little fragile depending on the font used, but it is probably better than the program hanging